### ISSUE TITLE
Code Style option to prevent repeated property patterns

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn-analyzers#4420]
From https://github.com/dotnet/csharplang/issues/4120; currently this is accepted:
```cs
s is { Length: > 0, Length < 10 }
```
It would be nice to have an analyzer to flag those cases and suggest to use `and` patterns instead.

We need to turn it off by default (maybe only as a hint?) for back compat but I expect some would prefer not to allow it in their codebase.

**Comment by mavasani** [comment]
I am wondering if this should be a compiler warning as part of compiler warning wave - this is flagging incorrect use of new language construct, which seems to fall in same bucket as https://github.com/dotnet/roslyn/issues/46637. Tagging @jmarolf @sharwell @333fred

**Comment by alrz** [comment]
See dotnet/csharplang#4120 for discussion on the warning wave. tldr: there's no incorrect behavior so we should continue to accept it. therefore my suggestion for an analyzer+codefix so we can opt-in for a warning/error.

**Comment by mavasani** [comment]
> there's no incorrect behavior

@alrz Are you suggesting this is a pure code style suggestion? We use the following classification based on analysis kind:

1. Correctness issue w.r.t usage of language constructs - compiler warning wave
2. Correctness issue w.r.t usage of .NET APIs or generic API agnostic code quality improvement suggestion - CA analyzer implemented in this repo
3. Code style suggestion w.r.t usage of new language constructs or code style preferences - IDE analyzer in Roslyn repo.

I am going to move this issue back to Roslyn repo so it can be discussed in the IDE design meeting.
